In non-strict mode, accept any variable as a tactic reference. (#12254)
coq/coq: doc/changelog/05-tactic-language/12254-wit-ref-dyn.rst

- **Changed:**

>   The "reference" tactic generic argument now accepts arbitrary
>   variables in non-strict mode
>   (`#12254 <https://github.com/coq/coq/pull/12254>`_,

Reviewer: If the point is to include local (goal) variables in `reference` (see below), I would suggest:
```suggestion
  The "reference" tactic generic argument now accepts arbitrary
   variables of the goal context
  (`#12254 <https://github.com/coq/coq/pull/12254>`_,
```
The fact that it accepts them in strict mode would then just derive from the general statement.

Author: Will do.
